node/node.go

```diff
@@ -377,7 +377,9 @@ func (n *Node) GetESDTData(address, tokenID string, nonce uint64) (*esdt.ESDigit
 	return esdtToken, nil
 }
 
-func (n *Node) getTokensIDsWithFilter(filterFunc func(esdtData *systemSmartContracts.ESDTData) bool) ([]string, error) {
+func (n *Node) getTokensIDsWithFilter(
+	filterFunc func(tokenIdentifier string, esdtData *systemSmartContracts.ESDTData) bool,
```

Comment:
this can be a function in the node. Large anonymous functions looks bad and are generally harder to test

Reply:
refactored